tcmur_cmd_handler.c

```diff
@@ -347,6 +341,10 @@ static int align_and_split_unmap(struct tcmu_device *dev,
 			tcmu_dev_dbg(dev, "There are totally %d splits\n", j);
 		}
 
+		pthread_mutex_lock(&state->lock);
+		state->refcount++;
+		pthread_mutex_unlock(&state->lock);
+	
```

Comment:
I moved this above the async_handle_cmd since the refcount is needed if the unmap_state_put is called before we get the ret from async_handle_cmd call. I then decrement it back down if the async_handle_cmd fails so that we dont have the extra refcount. This is the only change I made to your patch. I have tested and things work.

Reply:
That's not needed because the caller has a refcount on it already. It would not go down to zero until after we return from align_and_split_unmap and the caller releases their refcount.

Reply:
You mean the refcount is initialized to 1 thus we already have a refcount? So then we would want to change if(--state->refcount > 0) to if(--state->refcount => 0) ? TCMU-Runner crashes in my setup atm without the change to refcount++ prior to calling the async_handle_cmd

Reply:
Oh no, wait you are right about needing to do refcount++ before calling async_handle_cmd.

Reply:
We are ok on your side then right? It works for me now too.

Reply:
yes, I am good if you are good!

Reply:
Thanks!